flask/src/main/java/net/corda/gradle/flask/FlaskJarTask.java

```diff
@@ -0,0 +1,239 @@
+package net.corda.gradle.flask;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import net.corda.flask.common.Flask;
+import net.corda.flask.common.ManifestEscape;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.internal.file.CopyActionProcessingStreamAction;
+import org.gradle.api.internal.file.copy.CopyAction;
+import org.gradle.api.internal.file.copy.CopyActionProcessingStream;
+import org.gradle.api.internal.file.copy.FileCopyDetailsInternal;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.plugins.BasePluginConvention;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.WorkResult;
+import org.gradle.api.tasks.bundling.AbstractArchiveTask;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.security.MessageDigest;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class FlaskJarTask extends AbstractArchiveTask {
+
+    @Setter
```

Comment:
ok, this makes sense since Groovy is able to override property access as in this example

```groovy
class Dyn {
    private final Map<String, Object> __dict__ = new HashMap<>()
    def getProperty(String key) {
       return __dict__.get(key) 
    }
    
    void setProperty(String key, Object value) {
        __dict__.put(key, value)
    }
}

Dyn dyn = new Dyn()

dyn.name = "John"
dyn.surname = "Doe"
dyn.age = 42

println(dyn.name)
//Prints "John"
```

but I wonder if also need a getter method (which is missing here) for it to be usable from kotlin DSL (which cannot avail of Groovy magic)

Reply:
These properties _do_ have "getter" methods, e.g.:
```java
@Input
public Property<String> getLauncherClassName() {
    return launcherClassName;
}
```

Reply:
yeah, I forgot about those.. will fix